src/main/java/googlesignin/GoogleUserMapper.java

```diff
@@ -7,16 +7,22 @@
 import java.net.URLConnection;
 import java.util.Base64;
 
-public class GoogleUserMapper {
-    public String getUsernameFromEmail(String email) {
+public final class GoogleUserMapper {
+    private static Base64.Encoder encoder;
+
+    private GoogleUserMapper() {
+        encoder = Base64.getEncoder();
+    }
```

Comment:
This is incorrect.
`encoder` is a static member but is initialised whenever an object of the class is created and not before that.
This is error prone.
If someone calls one of the static methods before an object is created, `encoder` will be set to null and a `NullPointerException` will be thrown.

You could instead do:
```
private static final Base64.Encoder encoder = Base64.getEncoder();

private GoogleUserMapper() {}
```

Or rather (preferred), leave `encoder` to be a local variable in the method `getPictureFromUrl`. `Base64.getEncoder` is a "fast" method in the sense that it does not create new objects for each call but returns the same object. So calling it in that method is safe and efficient, you need not store it in a static field here.

Reply:
Done.